packages/product/src/repositories/product.ts

```diff
@@ -121,7 +125,34 @@ export class ProductRepository extends AbstractBaseRepository<Product> {
       return manager.create(Product, product)
     })
 
-    await manager.persist(products)
+    await manager.persistAndFlush(products)
+
+    return products
+  }
+
+  async update(
+    data: WithRequiredProperty<ProductServiceTypes.UpdateProductDTO, "id">[],
+    context: Context = {}
+  ): Promise<Product[]> {
+    const manager = (context.transactionManager ??
+      this.manager_) as SqlEntityManager
+
+    const products = await Promise.all(
+      data.map(async (updateData) => {
+        if (!isDefined(updateData.id)) {
+          throw new MedusaError(
+            MedusaError.Types.NOT_FOUND,
+            `"id" must be defined`
+          )
+        }
```

Comment:
comment: This is already checked in the module service - should we remove it from here?

Reply:
I added this in here because when testing from a product service level and not the product module service level, we get a mikroORM error. Remove this still?  

<img width="674" alt="Screenshot 2023-07-12 at 10 35 57" src="https://github.com/medusajs/medusa/assets/5105988/5b9ea5a6-14a4-4124-be42-6c069c253c66">

Reply:
Yeah, I would say it's fine to remove. The product service will be internal to the module. Think validation should happen in public APIs